profile page && styled-components grid

maria-vslvn opened the pull request:
# Summary
Fixes #988 

*High-level description of what your changes are accomplishing*
Profile view implemented
Custom common components created in order to have less small components and avoid duplication of code. Check it please, maybe we could use it somehow or find a better decision?

  # To Test

1. Visit profile page, please notice: all data are mocked
2. as new global styles were applied, it need testing to

elena-zh commented:
Hey @maria-vslvn , great job. 

Some notes: 
1. bottom elements are missing on newly designed pages. It would be nice to add them 
![image](https://user-images.githubusercontent.com/70885163/131844653-93a958bb-6ec4-4883-8e23-55c7320ae2eb.png)

2. It seems to me, that background is blurred more than on implemented pages: 
![image](https://user-images.githubusercontent.com/70885163/131848404-d78d22b5-a252-4a8b-8339-5dc7bf00d1a6.png)

3. Also, I have a strange impression when I switch between Swap and profile pages, and see a clear background picture and a blurred one. May be we could leave the same background for all pages.. WDYT, @alfetopito ?
https://drive.google.com/file/d/1zey7wouL_qzBK1wHwinJQlCRUaFj4X6I/view

4. I see a lot of 'style' errors in the console
![image](https://user-images.githubusercontent.com/70885163/131848892-47c91370-76eb-4f89-9ac3-ad949c518af6.png)

5. I think, we should keep 'Copy' icon right after a referral link and not to move it to the next row
![image](https://user-images.githubusercontent.com/70885163/131849060-c712f618-e7c9-4b69-823a-797a3d9c3107.png)

6. Copy icon looks different as on other places in the app. Its behavior is also different:
![image](https://user-images.githubusercontent.com/70885163/131849173-9c07adc3-8f05-47be-b159-75a682ff5f67.png)
![image](https://user-images.githubusercontent.com/70885163/131849238-122d5a1c-3764-4d12-bebd-496f99969f26.png)
![image](https://user-images.githubusercontent.com/70885163/131849267-25abe9cb-af90-4cbc-8185-5cb111281cbf.png)

elena-zh commented:
Mobile: 
1. There is no an announcement at the top of the profile page: area seems to be empty in a mobile device
![image](https://user-images.githubusercontent.com/70885163/131851300-426675f8-9220-4e23-9ac9-cba96b2d2c58.png) 
![image](https://user-images.githubusercontent.com/70885163/131851320-412facd1-d481-41ef-ace4-65df21ce1c32.png)

2. On the mock-up all profile page elements are visible without scrolling (case about making elements smaller)
![image](https://user-images.githubusercontent.com/70885163/131851470-42f38a4a-f351-4c25-80db-6a87b2cb7385.png) 
![image](https://user-images.githubusercontent.com/70885163/131851503-9b83c0a1-d03b-4f42-b39d-0b4138f77751.png)

elena-zh commented:
@alfetopito , @alongoni , 
Global question: 
we have different bg color on existing content section and on the mock up. Moreover, the section looks transparent a bit on the mock-up. Do we need to apply all these changes to existing pages?
![image](https://user-images.githubusercontent.com/70885163/131856537-7e2a80f3-b1d6-4d31-bf86-d475838050c6.png)

alfetopito commented on src/assets/styles/styled.ts:
```diff
@@ -0,0 +1,268 @@
+  ${(props) => props.flexCol && 'display:flex;flex-direction:column;'}
+  ${(props) => props.flexRow && 'display:flex;flex-wrap:wrap; align-items:center'}
+  ${(props) => props.stretch && 'height:100%;'}
+  @media (max-width:767px) {
```
For this we usually have these media queries https://github.com/gnosis/cowswap/blob/b366fa833887f6a86dacf93f425e2dc6f250c811/src/theme/styled.ts#L70

W3stside replied:
yes please review the theme inside custom/theme and src/theme

biocom replied:
Commented elsewhere as well. We use this format:

```
${({ theme }) => theme.mediaWidth.upToSmall`
  // stuff
`};
```

maria-vslvn replied:
done

alfetopito commented on src/components/icons/index.tsx:
```diff
@@ -0,0 +1,76 @@
+  width: PropTypes.string,
+}
+
+export const CopyIcon: FunctionComponent<IconProps> = ({ color, width = '16' }): React.ReactElement => {
```
As Elena mentioned, there's already another copy icon already used in the app, I would prefer to re-use that to keep the style consistent.

maria-vslvn replied:
agree, replaced with existing component

alfetopito commented on src/components/icons/index.tsx:
```diff
@@ -0,0 +1,76 @@
+  width: PropTypes.string,
+}
+
+export const QuestionIcon: FunctionComponent<IconProps> = ({ color, width = '16' }): React.ReactElement => {
```
Same point as the copy icon. There is already another question icon already used in the project

maria-vslvn replied:
agree, replaced with existing component

alfetopito commented on src/custom/pages/App/AppMod.tsx:
```diff
@@ -82,13 +100,22 @@ const Marginer = styled.div`
+  const [bgBlur, setBgBlur] = useState(false)
+  const location = useLocation()
+  useEffect(() => {
+    if (location.pathname.length > 1 && location.pathname !== '/swap') {
+      setBgBlur(true)
+    } else {
+      setBgBlur(false)
+    }
```
Refactoring suggestion:

```suggestion
    setBgBlur(location.pathname.length > 1 && location.pathname !== '/swap')
```

Although, not sure should keep this logic. I'll wait for everyone else's input regarding the blur on swap page.

anxolin replied:
@maria-vslvn i think Leandro suggestion is good

alfetopito commented on src/custom/pages/Profile/index.tsx:
```diff
@@ -1,87 +1,148 @@
+            Your referral url
+          </Txt>
+          <Wrap alignItems={'center'} justifyCont={'center'}>
+            <Txt onClick={() => copyToClipboard()} fs={14} txtAlign={'center'}>
```
There's a `<Copy>` component for that you can use. 
It's https://github.com/gnosis/cowswap/blob/7500644d17d8021fed99f218abf6582fde2ac7a1/src/custom/components/Copy/CopyMod.tsx#L31. 
Import it as `import Copy from 'components/Copy'`

maria-vslvn replied:
applied

alfetopito commented on src/custom/theme/baseTheme.tsx:
```diff
@@ -53,6 +53,9 @@ export function colors(darkMode: boolean): Colors {
     blueShade2: '#011e34',
     border: darkMode ? '#3a3b5a' : 'rgb(58 59 90 / 10%)',
     border2: darkMode ? '#254F83' : '#afcbda',
+    cardBorder: darkMode ? '#021E34' : 'rgba(255, 255, 255, 0.5)',
+    cardShadow21: darkMode ? '#4C7487' : '#FFFFFF',
+    cardShadow22: darkMode ? 'rgba(1, 10, 16, 0.15)' : 'rgba(11, 37, 53, 0.93)',
```
Any reason for using `21` then `22`?
Maybe you meant to have `1` and `2` ? Or no index and `2`?

maria-vslvn replied:
yes, at first i thought about using indexes and forgot to remove it later. 2-index removed

alfetopito commented on src/custom/theme/cowSwapTheme.tsx:
```diff
@@ -111,9 +115,9 @@ function themeVariables(darkMode: boolean, colorsTheme: Colors) {
-      borderRadius: '16px',
-      border: `3px solid ${colorsTheme.black}`,
+      boxShadow: `none`,
+      borderRadius: '1rem',
```
I suspect @biocom will say: "use `px` instead of `rem` for consistency with the rest of the app"

biocom replied:
Yes.

alfetopito reviewed (commented):
Looking good so far!

I left comments regarding the code, with a few overlaps with Elena's comments.
I'll ping the rest of the team to double check that part.

Now going over Elena's points:

1. Bottom elements missing: 
Agree, they should be visible. Probably need z-index adjustment or something?
2. Design is more blurred than implementation: 
We can make it more blurry, but I find it ok as is.
3. Swap page is not blurred while everything else is: 
Good point, we have to decide on the behaviour here.
I'm not that concerned for not being blurred on the swap page. I even makes sense, to have the main page not blurred, and on other pages you add a blur to indicate "secondary" or something.

My concern here is not having the swap border:
| before | now |
| - | - |
![screenshot_2021-09-02_13-28-19](https://user-images.githubusercontent.com/43217/131911538-60e62b45-cb72-4883-86d3-f4326f4ede4d.png) | ![screenshot_2021-09-02_13-17-24](https://user-images.githubusercontent.com/43217/131911558-6c2f4570-f2db-4f28-8f12-5e618e562f57.png)

I'm not saying we should add it back, just saying we probably need some way to emphasize the swap form.
Designers?

4. `Style` errors on the console: Agree
5. Copy icon wrapping: Agree
6. Copy icon style/behaviour: Agree
7. Cowgame score on mobile:
Just checked and the same happens on PROD, so not an issue introduced here.
8. Profile page width larger than design:
Not a big deal to me, I'll leave this up to the designers
9. Text colors on dark mode:
Same as previous

Regarding mobile points:

1. No announcement visible on top: 
Same as point `1` above, probably need to adjust z-index
2. No all items viewable in a single page: 
Agree, it could be make slight smaller and reduce the margin between cards


Regarding global question:

1. Different colors between design and current backgrounds. Also there's transparency in the design:
Hawk eyes! Another one that I leave up to the designers

alongoni commented on src/custom/pages/App/AppMod.tsx:
```diff
@@ -35,14 +35,32 @@ import ReferralLinkUpdater from 'state/affiliate/updater'
+    left: 0;
+    bottom: 0;
+    right: 0;
+    filter: blur(5px);
```
Maybe:
filter: blur(10px);
cc @biocom?

biocom replied:
20px is better.

alongoni reviewed (commented):
Great job Maria!

W3stside commented on src/assets/styles/styled.ts:
```diff
@@ -0,0 +1,268 @@
+  >
+>`
+  display: inline-flex;
+  text-align: ${(props) =>
+    props.txtAlign ? (typeof props.txtAlign !== 'string' ? props.txtAlign[0] : props.txtAlign) : 'left'};
+  font-size: ${(props) => (props.fs ? (typeof props.fs !== 'number' ? props.fs[0] / 16 : props.fs / 16) : 0.87)}rem;
+  opacity: ${(props) => props.secondary && '0.5'};
+  line-height: ${(props) =>
```
Hi @maria-vslvn yes this css file is much too overcomplicated imho.

Why add so many customisable props? why not use inheritance on a base component or simply create ones you need for the job? While i think it's nice to aim to make something generic and highly customisable, in this case the code is very difficult to read and overly verbose.

i would stick with smaller, more simple components here please.

biocom replied:
Please simplify the code. Use px's instead of rem.

maria-vslvn replied:
done

anxolin reviewed (commented):
Nice to see the profile page!

In general, agree with most comments. Just a couple of comments. I feel needs a bit of refinement in general. 

>> Swap page is not blurred while everything else is:
> Good point, we have to decide on the behaviour here.
> I'm not that concerned for not being blurred on the swap page. I even makes sense, to have the main page not blurred, and > on other pages you add a blur to indicate "secondary" or something.

Agree with @alfetopito . I find it ok, to have the main page non-blured and content pages with the effect.

Regarding the images, they look not very good quality. Not sure if it's the size, resolution, or small details.

Also, are the copy button, has a background only in light mode. And doesn't have any hover effect or cursor. 

![image](https://user-images.githubusercontent.com/2352112/131985748-2cbb11ec-d2a9-43b5-a155-fdda8bac0446.png)

Since it's the only button in the page, I wonder if it should look like the other buttons? 
![image](https://user-images.githubusercontent.com/2352112/131985921-c365d0ba-1ff0-439c-aec9-f0d94c633e63.png)

The copy URL also should be dynamic

![image](https://user-images.githubusercontent.com/2352112/131986018-948e806f-32a1-40d9-af61-c231f35994dd.png)

biocom commented:
Here my review:
- For the 'click to copy' please see the sidebar click logic/style. Would be good to match the style + behavior (e.g. hover cursor).
- The Emoji images are intended to be actual emoji's. So use the actual emoji unicode characters -> 👨‍🌾🐮 etc.
- The blur effect was intended to be on all pages, except the Swap page. Primarily to function as a content page differentiator.
- For now don't modify the main Swap container's styles.

elena-zh commented:
> Cowgame score on mobile:
> Just checked and the same happens on PROD, so not an issue introduced here.
 My bad.. I have created a separate issue for this case #1388

elena-zh commented:
I think, I missed one important thing to mention: Profile page is does not correspond to the mock-up to a not connected user: 
 - Connect to a wallet button should be added to the bottom of the section. The button should be enabled and should call the connection modal
 - there should not be 'Last updated' field near the page header
![image](https://user-images.githubusercontent.com/70885163/131993650-6e21f1aa-810a-4ed5-b331-64a43940b7ae.png)

elena-zh commented:
As for 'Style' issue in the console, I can face it prod, but not so often as here. 
However, I created a separate issue for this case #1389

alongoni commented:
hey Maria! Looks great. Some comments:
1)  The title of the page "Profile overview" is 32px on mobile. Maybe we should decreases it to 24px. It could affect all pages.
![image](https://user-images.githubusercontent.com/622217/133137628-ad12ef0c-2ad9-47cc-a46b-8984564e77f9.png)
2) [Titles color on dark-mode](https://www.figma.com/file/bcoZf5RIgVKDItdOaTZq18/GP-Swap?node-id=2647%3A289). Titles should have #C5DAEF (or the equivalent color variable). Also I've made some little colors adjustments in light mode in order to have more consistent between light and dark mode (see comments on Figma).

alfetopito commented:
Looking quite nice!

Regarding the swap page though, I think it shouldn't have a transparent background
![screenshot_2021-09-20_15-17-00](https://user-images.githubusercontent.com/43217/134083958-5f2fdaf3-3ee9-42c0-a8ef-d36fc2204fb1.png)
In my view doesn't look as good as other pages since the background is not blurred.

Also, please don't forget to address the comments and mark as resolved the ones you already handled.

biocom commented:
Agreed with @alfetopito 's comments. I'd leave the Swap page/container as is.

elena-zh commented:
I have retested changes and here is the list of remaining issues to fix (besides mentioned above ones):
1. Page components are shifted to a not connected user. Moreover, 'Referral' section is not displayed in a mobile view now: 
![not connected](https://user-images.githubusercontent.com/70885163/134170052-d0a521e7-25b3-4d47-8f6e-473ae6470be5.jpg)
![not connected - missing](https://user-images.githubusercontent.com/70885163/134170067-38c9bb82-f408-4e74-bd2c-a5a4dae69e02.jpg)

2. Last updated field should be removed from a profile page when a user is not connected
![image](https://user-images.githubusercontent.com/70885163/134170169-d18363da-24c4-4f88-bdb4-61b7842636b1.png)

3. Connect to a wallet button should not present when a user is connected to a wallet
![connect](https://user-images.githubusercontent.com/70885163/134170299-5bcdb298-ca0e-465b-9f65-a4f99d0b5230.jpg)

4. 'Connect to a wallet' button style is different from the button's style at the top of the page.
![image](https://user-images.githubusercontent.com/70885163/134170463-925460b4-8c81-4543-a7f9-d1606aaf3e81.png)
I would reuse it, IMO
![re-use](https://user-images.githubusercontent.com/70885163/134170510-22bf9131-31dd-4961-9290-b0833245ff5f.jpg)
![re-use2](https://user-images.githubusercontent.com/70885163/134170525-95f0292c-39b5-435d-b90d-63f0acaf9d35.jpg)

5. Copy icon has a circle with bg color, however, original icon from the side bar does not have one: 
![copy bg](https://user-images.githubusercontent.com/70885163/134170751-64936a5f-59d5-42cd-8a28-3990082e0a55.jpg)
![copy bg2](https://user-images.githubusercontent.com/70885163/134170766-43874f3b-1258-4d74-be08-e1333d13cde8.jpg)

6. Copy icon should be bottom-aligned in a mobile view
![copy to bottom](https://user-images.githubusercontent.com/70885163/134170865-e76b9486-5ec7-44bb-b01b-221f7b0a6526.jpg)

7. Still, announcement area is empty at the top of the page
8. 'Style' error in console is not fixed
![ыенду](https://user-images.githubusercontent.com/70885163/134171057-ccdda425-2573-4de8-a9fc-b7c9d3478094.jpg)
9. Still, it would be nice to use smaller elements in a mobile view
10. Microsoft icons look.. strange =) However, I'm not sure whether we should fix it. Just to mention
![microsoft icons](https://user-images.githubusercontent.com/70885163/134171287-6fb7603d-9e79-44f3-a9f0-c2a90122ae45.jpg)

maria-vslvn commented on src/assets/styles/styled.ts:
```diff
@@ -0,0 +1,268 @@
+}
+
+// with font size convert to rems
+export const Txt = styled.div<
```
agree, i used h3

maria-vslvn commented on src/custom/pages/Profile/index.tsx:
```diff
@@ -1,87 +1,148 @@
+                      style={{ margin: ' .625rem auto' }}
+                      src="https://emojipedia-us.s3.dualstack.us-west-1.amazonaws.com/thumbs/72/apple/118/handshake_1f91d.png"
+                      alt={'handshake_1f91d'}
+                    />
```
done

maria-vslvn commented on src/custom/pages/Profile/index.tsx:
```diff
@@ -1,87 +1,148 @@
+                      -
+                    </Txt>
+                    <Txt fs={14} secondary margin={'.5rem 0'}>
+                      Referrals Volume
```
done

maria-vslvn commented:
@elena-zh 
5. https://www.figma.com/file/bcoZf5RIgVKDItdOaTZq18/GP-Swap?node-id=2647%3A289
but there is a bg in mockups, so i added
8. cant see style ers in console, could you please provide repro steps?
9. made as small as could, you can give me any more advices on what else to make smaller
10. that's just how they look ¯\_(ツ)_/¯

alongoni commented:
> @elena-zh
> 5. https://www.figma.com/file/bcoZf5RIgVKDItdOaTZq18/GP-Swap?node-id=2647%3A289
> but there is a bg in mockups, so i added
> 8. cant see style ers in console, could you please provide repro steps?
> 9. made as small as could, you can give me any more advices on what else to make smaller
> 10. that's just how they look ¯_(ツ)_/¯

Great @maria-vslvn  thanks!
9) I think @elena-zh refers to this case, right? (medium-screens). Maybe we can change the breakpoint and maintain the desktop appearance for medium devices.
![Screenshot from 2021-09-23 10-39-56](https://user-images.githubusercontent.com/622217/134517889-8ec26756-a95b-4170-8f5c-e7977af97c11.png)
In small screens looks great!
10) The icons displayed may depend on the operating system. I see this icons on Chrome using Linux. Maybe we need to export them in svg to have consistency across all platforms.
![Screenshot from 2021-09-23 10-55-22](https://user-images.githubusercontent.com/622217/134520245-d70090ad-f211-4b92-988a-8856e6cee5f9.png)

maria-vslvn commented:
@elena-zh  @alongoni  added the upd about decreasing the brekpoints

elena-zh commented:
Thanks!!
Some new issues + some unfixed: 
1. Still missing the Black boarder on the Swap form
![black border](https://user-images.githubusercontent.com/70885163/134529144-ded6ed67-d31b-4b7d-a58f-fd3489412850.jpg)
2. Profile page is missing in the current PR. As well as 'About' section is removed from the menu
![profile](https://user-images.githubusercontent.com/70885163/134529285-3d9e31ad-7530-404c-b7bc-5aacb00dde79.jpg)
3. Connection modal is broken: this is how it looks like now
![connect dark](https://user-images.githubusercontent.com/70885163/134529698-21e07dc9-6161-4872-9f43-f96906a5aa3c.jpg)
![connect light](https://user-images.githubusercontent.com/70885163/134529710-760539f1-b613-432a-9dc7-1f93afbb4042.jpg)
**Expected**: 
![expected dark](https://user-images.githubusercontent.com/70885163/134529758-e7930da2-8609-40cf-b2f8-964e50b480bc.jpg)
![expected lignt](https://user-images.githubusercontent.com/70885163/134529775-cfbf235e-78e0-466a-9a06-c023477a8f71.jpg)
4. When a screen width is less than 375 px, elements exceed screen area. Look: swap page is adjusted when it is displayed on the 320 px width, Profile page - not
![profile - small screen](https://user-images.githubusercontent.com/70885163/134530162-2fbbf821-5b2e-4bfc-a6a3-a9c79212a596.jpg)
![swap](https://user-images.githubusercontent.com/70885163/134530210-2ed20388-18ca-4b4c-ab4c-918e18767b68.jpg)
5. Moreover, elements are shifted when copy referral link
https://user-images.githubusercontent.com/70885163/134530635-842a683d-7b5c-4771-bfcc-e7d20a5698a0.mp4
6. Copy icon still has background circle
 ![copy bg](https://user-images.githubusercontent.com/70885163/134170751-64936a5f-59d5-42cd-8a28-3990082e0a55.jpg)
![copy bg2](https://user-images.githubusercontent.com/70885163/134170766-43874f3b-1258-4d74-be08-e1333d13cde8.jpg)
7. It would be nice to move copy icon to be displayed right after URL . Or, if it impossible, to show it at the left pot corner.. @alongoni , @alfetopito , WDYT?
8. Tooltips: according to the discussion https://github.com/gnosis/cowswap/issues/958#issuecomment-906314157 we do not need tooltips, I guess. @alfetopito , correct?

Thanks

alfetopito commented:
Replying to Elena's comments:

1. Yes, that's still missing. Swap UI should not be changed
2. Well, this is a constant change that has to be disabled on every release. I think it's fine not having it until we are ready to release the profile page, otherwise it might get removed again
3. Yep, should be like you said
4. Yep. For that either fix the base container to the cards minimum size or make the card container float. What's important is to not let the inner cards go over the outer container
5. Yes, it shifts a little, although I consider this a minor thing
6. Yep, should match the design
7. I don't get what you mean here Elena. The copy icon is already after the URL, no?
8. Tooltips can be added as a separated PR/issue

alfetopito commented on src/custom/pages/Profile/styled.tsx:
```diff
@@ -0,0 +1,136 @@
+
+  max-width: 910px;
+  min-height: auto;
+  padding-top: 1rem;
```
Use px instead of rem :)

alfetopito reviewed (commented):
Code is much better, great improvement!
Minor comment regarding the units

elena-zh commented:
> > It would be nice to move copy icon to be displayed right after URL . Or, if it impossible, to show it at the left pot corner.. @alongoni , @alfetopito , WDYT?

> I don't get what you mean here Elena. The copy icon is already after the URL, no?

@alfetopito , I'm sorry, I forgot to attach the screenshot:
![move copy](https://user-images.githubusercontent.com/70885163/134644574-211a5326-82d7-4af1-83d9-c3762dc48f54.jpg)

alongoni commented:
hey @maria-vslvn 
Regarding 6) I've[ update the design](https://www.figma.com/file/bcoZf5RIgVKDItdOaTZq18/GP-Swap?node-id=1989%3A2) and remove the background for [copy button](https://feathericons.com/?query=copy).

maria-vslvn commented:
1. fixed
2.  - 
3.  the modal incorrect look connected to the Connect button component reusability. if i should try fix it in this pr or use some separat button for that, wdyt @elena-zh @ramirotw ?
4. fixed
5. it was how it worked previously, if i should fix it here let me know or maybe we should create a ticket in order to avoid a lot of code in one pr and simplify its review
6. fixed

anxolin reviewed (commented):
> 3. the modal incorrect look connected to the Connect button component reusability. if i should try fix it in this pr or use some separat button for that, wdyt @elena-zh @ramirotw ?

That's fine with me, since this error doesn't affect to the main connect button and this page is hidden, is see no harm in merging and doing it in a different PR.

I know why this is happening, and I will create a small PR fixing the issue just for giving the direction.

anxolin commented:
The PR i mentioned i my previous PR: @alfetopito @ramirotw could someone in protofire work on this? https://github.com/gnosis/cowswap/pull/1477

elena-zh commented:
@maria-vslvn , great job!
I'm not able to test changes in this PR, as I get 404 error on opening it, but i was able to test changes in #1477 pr. 
1. The black boarder is back in the Swap page, however, it is back to the About/FAQ/Terms pages, As far as I understand, only Swap page should have black boarder on its area. All the rest pages (Profile/About/FAQ/Terms) should look the same 
2. I have created a separate issue for the case 5: #1478
3. Still I'd vote for moving 'Copy' icon closer to the referral link 
![image](https://user-images.githubusercontent.com/70885163/134914190-b522d1e8-a1cd-4c14-bce5-c85dbcee106f.png)
4. Btw, @alfetopito , @alongoni , should we have some kind if animation when press on the 'Last updated' icon? I know that it is not related to this PR, but just a question.
![image](https://user-images.githubusercontent.com/70885163/134914401-85e920d3-6958-4cb7-8510-fc8c70d2de6f.png)

elena-zh commented:
I have retested changes in the current PR after deployment error fix. All the mentioned above points are valid for this PR.

alongoni commented:
> 4. Btw, @alfetopito , @alongoni , should we have some kind if animation when press on the 'Last updated' icon? I know that it is not related to this PR, but just a question.
>    ![image](https://user-images.githubusercontent.com/70885163/134914401-85e920d3-6958-4cb7-8510-fc8c70d2de6f.png)

I think it is just an icon with automatic update. Maybe we can add a transition each time it runs.

alfetopito commented:
>1. The black boarder is back in the Swap page, however, it is back to the About/FAQ/Terms/404 pages, As far as I understand, only Swap page should have black boarder on its area. All the rest pages (Profile/About/FAQ/Terms/404) should look the same

Yep. But I believe we are in a state that can be merged and address the remaining points in follow up PRs
 
>2. I have created a separate issue for the case 5: [Mobile: Profile elements are shifted when copy referral link #1478](https://github.com/gnosis/cowswap/issues/1478)
> 
>3. Still I'd vote for moving 'Copy' icon closer to the referral link
>        ![image](https://user-images.githubusercontent.com/70885163/134914190-b522d1e8-a1cd-4c14-bce5-c85dbcee106f.png)

Agree. Maybe that'll help solve the issue with mobile and the text shifting mentioned on `3`. Then again, another PR/issue

>4. Btw, @alfetopito , @alongoni , should we have some kind if animation when press on the 'Last updated' icon? I know that it is not related to this PR, but just a question.
>![image](https://user-images.githubusercontent.com/70885163/134914401-85e920d3-6958-4cb7-8510-fc8c70d2de6f.png)

No, this is intended to be a static icon. Maybe in a future version we could have that, but right now the button is not intended to be clickable.

alfetopito reviewed (approved):
I quite like the result and I believe this is already "merge-able"

There are a few things to address yet but they can be addressed in smaller PRs.
Also, some of the issue are not entirely UI specific, which is out of the scope of Maria's task.

With that in mind will be easier for reviewers to comment

When CSS/design people (@alongoni @biocom) are happy let's merge it.

@maria-vslvn in the mean time, you can start addressing the pending issues in PRs pointing to this branch.

biocom commented:
Minor thing i missed:
- The profile container misses this setting `backdrop-filter: blur(50px);`

Otherwise OK to go as is.

elena-zh commented:
I have created separate issues for moving copy closed to the link (#1480) and boarder color fix (#1481)
